### PR TITLE
pm: refresh ticket params before expiry from sender side

### LINKED ISSUE #1852: Refresh session when within the last 10% of ticket params validity period
**Is your feature request related to a problem? Please describe.**

At the moment, B will refresh a session (i.e. call GetOrchestrator on O to get a fresh OrchestratorInfo message) if the O's [ticket params are expired](https://github.com/livepeer/go-livepeer/blob/7a0fd1f6b958d61177fc01c488972f54b8bad1e3/server/broadcast.go#L854). However, ticket params can expire while a segment is in flight to O. If this happens, the O will return a TicketParams expired error when the segment is received and B will temporarily suspend O for returning an error.

**Describe the solution you'd like**

B should refresh a session if O's ticket params are *almost* expired. B refreshes a session if O's auth token is within 10% of the [end of its validity period](https://github.com/livepeer/go-livepeer/blob/7a0fd1f6b958d61177fc01c488972f54b8bad1e3/server/broadcast.go#L845). We can do something similar with ticket params where B refreshes a session if O's ticket params are within 10% of the end of its validity period.

### PR DESCRIPTION
**Pull request by kyriediculous**
**What does this pull request do? Explain your changes. (required)**
Refresh TicketParams for the active orchestrator session on the broadcaster at 10% before they expire. 

**Specific updates (required)**
- Updated the expiry check in `sender.ValidateTicketParams`

**How did you test each of these updates (required)**
- Added unit tests

**Does this pull request close any open issues?**
Fixes #1852 

**Checklist:**
- [x] Read the [contribution guide](./doc/contributing.md)
- [x] `make` runs successfully
- [x] All tests in `./test.sh` pass
- [x] README and other documentation updated
- [x] [Pending changelog](./CHANGELOG_PENDING.md) updated

### CODE REVIEW

File: pm/recipient.go
Comment (yondonfu) [inline pm/recipient.go]: A problem I see with this method of calculating a buffer before the expiration is that it explicitly depends on `paramsExpirationBlock` which is determined by O. While this is a hardcoded variable, there is no guarantee that B & O will have the same view of the value of this variable since they could be running different versions of go-livepeer [1]. For example, it could be the case that after #1873 `paramsExpirationBlock = 10` on B, but an O is still using an older version of go-livepeer where `paramsExpirationBlock = 5`).

What if we just return `ErrTicketParamsExpired` if we are within 1-2 blocks of the expiration block?

[1] I realize that this is also a problem for the [auth token](https://github.com/livepeer/go-livepeer/blob/master/server/broadcast.go#L846) buffer calculation

Comment (kyriediculous) [reply pm/recipient.go]: fixed in 2a26c888

Comment (kyriediculous) [reply pm/recipient.go]: > [1] I realize that this is also a problem for the auth token buffer calculation

Want to address this in a separate issue ?

File: pm/sender.go
Comment (yondonfu) [inline pm/sender.go]: Nit: If we're defining a buffer as the # of blocks before the expiration block could the following be slightly more readable?

```golang
currentBuffer := new(big.Int).Sub(ticketParams.ExpirationBlock, latestBlock).Int64()
if currentBuffer <= paramsExpiryBuffer {
    ...
}
```

Otherwise, the current code evaluates to a negative buffer in the normal case and only evaluates to a positive buffer when `ErrTicketParamsExpired` should be returned.

File: pm/recipient.go
Comment (yondonfu) [reply pm/recipient.go]: > Want to address this in a separate issue ?

👍

**Review by yondonfu — APPROVED** [review]
LGTM after squashing & fixing merge conflict.

**Pull request merged**
